controller/array_action/array_mediator_svc.py

```diff
@@ -626,29 +634,32 @@ def delete_snapshot(self, snapshot_id):
     def get_host_by_host_identifiers(self, initiators):
         logger.debug("Getting host name for initiators : {0}".format(initiators))
         detailed_hosts_list = self._get_detailed_hosts_list()
-        iscsi_host, fc_host = None, None
+        iscsi_host, nvme_host, fc_host = None, None, None
+        connectivity_types = set()
         for host in detailed_hosts_list:
             if initiators.is_array_iscsi_iqns_match(host.iscsi_names):
                 iscsi_host = host.name
+                connectivity_types.add(config.ISCSI_CONNECTIVITY_TYPE)
                 logger.debug("found iscsi iqn in list : {0} for host : "
                              "{1}".format(initiators.iscsi_iqn, iscsi_host))
             if initiators.is_array_wwns_match(host.wwns):
                 fc_host = host.name
+                connectivity_types.add(config.FC_CONNECTIVITY_TYPE)
                 logger.debug("found fc wwns in list : {0} for host : "
                              "{1}".format(initiators.fc_wwns, fc_host))
-        if iscsi_host and fc_host:
-            if iscsi_host == fc_host:
-                return fc_host, [config.ISCSI_CONNECTIVITY_TYPE,
-                                 config.FC_CONNECTIVITY_TYPE]
+            logger.debug("check initiator nqn : {0} vs host nqn : {1}".format(initiators.nvme_nqn, host.nqn))
+            if initiators.is_array_nvme_nqn_match(host.nqn):
+                nvme_host = host.name
+                connectivity_types.add(config.NVME_OVER_FC_CONNECTIVITY_TYPE)
+                logger.debug("found nvme nqn in list : {0} for host : "
+                             "{1}".format(initiators.nvme_nqn, nvme_host))
+        if not connectivity_types:
+            logger.debug("can not found host by using initiators: {0} ".format(initiators))
+            raise array_errors.HostNotFoundError(initiators)
+        host_name = self.get_host_name_if_equal(iscsi_host, nvme_host, fc_host)
+        if not host_name:
             raise array_errors.MultipleHostsFoundError(initiators, fc_host)
-        if iscsi_host:
-            logger.debug("found host : {0} with iqn : {1}".format(iscsi_host, initiators.iscsi_iqn))
-            return iscsi_host, [config.ISCSI_CONNECTIVITY_TYPE]
-        if fc_host:
-            logger.debug("found host : {0} with fc wwn : {1}".format(fc_host, initiators.fc_wwns))
-            return fc_host, [config.FC_CONNECTIVITY_TYPE]
-        logger.debug("can not found host by using initiators: {0} ".format(initiators))
-        raise array_errors.HostNotFoundError(initiators)
+        return host_name, list(connectivity_types)
```

Comment:
why use `set` if we return a `list`?

Reply:
duplicates (multiple hosts with the same connectivity type)

Reply:
actually, I dont sure it's possible, I think its a UT error

Reply:
is it not possible to have multiple hosts with the same connectivity type (user misconfiguration)?